src/CSESoftware.Repository/IBulkRepository.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using CSESoftware.Core.Entity;
+
+namespace CSESoftware.Repository
+{
+    public interface IBulkRepository : IRepository
+    {
+        Task<IEnumerable<object>> CreateAsync<TEntity>(IEnumerable<TEntity> data) where TEntity : class, IBaseEntity;
```

Comment:
Should you make this return an IEnumerable<T> and specify that it is a TEntity<T>? Is that possible / does that make sense to do?

Reply:
The intent here isn't so much to return the whole entities that were inserted, rather the id's so you can requery if you need or continue FK operations. if we do a select * on millions of records im not sure we even speed anything up with the overhead that would add.